octobot_trading/data_adapters/candles_adapter.py

```diff
@@ -0,0 +1,95 @@
+
+
+def get_symbol_close_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_close_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_close_candles(limit)
+
+
+def get_symbol_open_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_open_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_open_candles(limit)
+
+
+def get_symbol_high_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_high_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_high_candles(limit)
+
+
+def get_symbol_low_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_low_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_low_candles(limit)
+
+
+def get_symbol_volume_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_volume_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_volume_candles(limit)
+
+
+def get_symbol_time_candles(symbol_data, time_frame, limit, include_in_construction):
+    tf = TimeFrames(time_frame)
+    if include_in_construction:
+        return _add_in_construction_data(
+            symbol_data.symbol_candles[tf].get_symbol_time_candles(limit),
+            symbol_data,
+            tf,
+            PriceIndexes.IND_PRICE_CLOSE.value)
+    else:
+        return symbol_data.symbol_candles[tf].get_symbol_time_candles(limit)
+
+
+def _add_in_construction_data(candles, symbol_data, time_frame, data_type):
+    # TODO: optimize numpy call if possible
+    return np.append(candles, symbol_data.symbol_klines[time_frame].kline[data_type])
```

Comment:
We need to think about it, but I had an idea to optimize this line.
We can modify shift with the numpy `shift` method like the `candle_manager` do when a new candle is added and then set the last candle with the kline data. What do you think about it ? My 2 main concerns here are : 
- We'll lost candle[0]
- The shift duration

Reply:
I think candle[0] doesn't matter, about time if it's faster than the current implementation, we should do it.

Reply:
It should be better to not recreate a new numpy array but I am not sure. Yes we should test it 👍